backend/src/main/java/ai/giskard/web/rest/errors/GalleryDemoSpaceException.java

```diff
@@ -0,0 +1,9 @@
+package ai.giskard.web.rest.errors;
+
+import org.springframework.http.HttpStatus;
+
+public class GalleryDemoSpaceException extends GiskardException {
+    public GalleryDemoSpaceException(String description) {
+        super(HttpStatus.SERVICE_UNAVAILABLE, description);
```

Comment:
I would suggest to use NOT_ALLOWED (https://www.rfc-editor.org/rfc/rfc7231#section-6.5.5) that seems more suitable for us.

SERVICE_UNAVAILABLE is more when the server is overloaded or down.

Reply:
Is it more generally used in the case where the HTTP method is not allowed?
I thought about adopting it thanks to our RESTful APIs, just worrying about that some APIs might also return NOT_ALLOWED when using an inappropriate HTTP method.

Reply:
Well the rule of thumbs is that 5xx means server error while 4xx means client error. Technically we also can have a 503 for other things.

In this case we can indeed have a 405 error when we try to `POST` a `GET` endpoint only. In our point it technically is the case as the `POST` is disabled when readonly if that makes sense. But subjectively it also could be a 403 error.

I think we can use the error message description to differenciate between different source (or response body)